feat(chatMessageHistory): add create method and test

Itzaprado opened the pull request.

Itzaprado committed:
feat(chatMessageHistory): add create method and test

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/chatMessageHistory.ts:
```diff
@@ -0,0 +1,86 @@
+import PostgresEngine from "./engine.js";
+
+export class PostgresChatMessageHistory extends BaseChatMessageHistory {
+  lc_namespace: string[];
```
```suggestion
  lc_namespace: string[] = ["langchain", "stores", "message", "google-cloud-sql-pg"];
```

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/chatMessageHistory.ts:
```diff
@@ -0,0 +1,86 @@
+    return new PostgresChatMessageHistory(engine, sessionId, tableName)
+  }
+
+  addUserMessage(message: string): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  addAIChatMessage(message: string): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
```
These helper methods are not needed and will be defined by the super class

Itzaprado replied:
Ok, but BaseChatMessageHistory is an abstract class and so are these methods, so they need to be implemented. If I remove them, I get an error message like this _"Non-abstract class 'PostgresChatMessageHistory' does not implement inherited abstract member 'addAIChatMessage' from class 'BaseChatMessageHistory'."_

averikitsch replied:
Got it. Let's continue to target BaseChatMessageHistory and implement these methods versus the similar BaseListChatMessageHistory

Itzaprado replied:
Ok, I will do it when I implement the addMessage method since it is not implemented yet.

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/chatMessageHistory.ts:
```diff
@@ -0,0 +1,86 @@
+  
+  schemaName: string;
+
+  constructor(engine: PostgresEngine, sessionId: string, tableName: string, schemaName: string = "public") {
```
It looks like a lot of the other integrations use an Inputs interface for the constructor, example https://github.com/langchain-ai/langchainjs/blob/0f9139d95f1f73e3cd260141f86c0e3ffc68de3d/libs/langchain-mongodb/src/chat_history.ts. Let's follow that pattern and turn the constructor input into a object

Itzaprado committed:
refactor: update chatMessageHistory interface
[ ] Change to an object the args received in the constructor
[ ] Add lc_namespace array

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/chatMessageHistory.ts:
```diff
@@ -2,8 +2,15 @@ import { BaseChatMessageHistory } from "@langchain/core/chat_history";
 import { BaseMessage } from "@langchain/core/messages";
 import PostgresEngine from "./engine.js";
 
+export interface PostgresChatMessageHistoryArgs {
```
```suggestion
export interface PostgresChatMessageHistoryInput {
```

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/chatMessageHistory.ts:
```diff
@@ -2,8 +2,15 @@ import { BaseChatMessageHistory } from "@langchain/core/chat_history";
+  engine: PostgresEngine;
+  sessionId: string;
+  tableName: string;
+  schemaName: string;
```
Set schemaName default to public

Itzaprado replied:
I can't define it in the interface declaration, but the schemaName is public by default in the create method and I just added it to the constructor as well.

Itzaprado committed:
refactor: change interface name and set default in constructor
